src/dependencies.jl

```diff
@@ -313,6 +313,53 @@ function provides{T}(::Type{T},providers::Dict; opts...)
     end
 end
 
+const have_sonames = Ref(false)
+const sonames = Dict{String,String}()
+reread_sonames() = (empty!(sonames); have_sonames[] = false; nothing)
+
+if is_windows() || is_apple()
+    function read_sonames()
+        have_sonames[] = true
+    end
+elseif is_linux()
+    let ldconfig_arch = Dict(:i686 => "x32",
+                             :x86_64 => "x86-64",
+                             :aarch64 => "AArch64"),
+        arch = get(ldconfig_arch, Sys.ARCH, ""),
+        fmt = Regex("^\\s+(\\S+)\\.so\\S+ \\([^)]*$arch[^)]*\\) => (.+)\$")
+    global function read_sonames()
+        empty!(sonames)
+        for line in eachline(`/sbin/ldconfig -p`)
+            VERSION < v"0.6" && (line = chomp(line))
+            m = match(fmt, line)
+            if m !== nothing
+                sonames[m[1]] = m[2]
+            end
+        end
+        have_sonames[] = true
+    end
+    end
+else
+    let fmt = Regex("^\\s:-l(\\S+)\\.\\S+\\s(.+)\$")
```

Comment:
How does it distinguish between something like `libLLVM-3.8.so.1` (with digits in the name) and `libperl.so.5.22` (with multiple digits in the version number)?